hazelcast-client-legacy/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -401,9 +401,9 @@ protected Object putInternal(final K key, final V value, final ExpiryPolicy expi
             }
         } else {
             if (nearCache != null || statisticsEnabled) {
-                future.andThen(new ExecutionCallback<Object>() {
```

Comment:
We don't want to use same thread. That thread is ResponseThread in this case. 
We want to run on user thread. And only way seems to be by overwriting future that we gave to user.